### ISSUE TITLE
Describe workflow as code

### ISSUE THREAD

**Issue by moritzzimmer** [wercker/wercker#269]
We currently use similar workflows in our wercker applications. It's kind of tedious and error prone that we have to wire the pipelines manually to workflows in the web UI.

Do you have any plans to provide a possibility to describe wercker workflows in code which are then picked-up automatically?

**Comment by hobti01** [comment]
I agree with the idea of all configuration within the wercker.yml for the following reasons:
- The split configuration of wercker.yml and Web UI is very confusing for new users and creates a painful and manual administration point.
- The requirement to create a pipeline in the wercker.yml and then create it again in the UI in order to execute it is not intuitive and not maintainable for many projects.
- Since the pipeline creation and workflow creation dialogs ask for such limited data, can the information be defined in wercker.yml - i.e. pipeline as code instead of pipeline as some code and some UI clicking?  Vice-versa: if the workflows/pipelines can be created in the UI, why not export/serialize them to wercker.yml to save into the codebase?

https://wercker.uservoice.com/forums/181077-general/suggestions/18637906-allow-all-pipeline-and-workflow-configuration-in-w
